src/web3/KeychainAccountManager.ts

```diff
@@ -263,24 +229,31 @@ export class KeychainSigner implements Signer {
     return true
   }
 
-  async decrypt(ciphertext: Buffer): Promise<Buffer> {
-    return this.localSigner.decrypt(ciphertext)
-  }
-
-  async computeSharedSecret(publicKey: string): Promise<Buffer> {
-    return this.localSigner.computeSharedSecret(publicKey)
+  /**
+   * Get the unlocked contractkit signer. Throws if not unlocked.
+   */
+  get unlockedContractKitSigner() {
+    if (!this.isUnlocked()) {
+      this.localContractKitSigner = null
+      this.localEthersWallets = new Map()
+    }
+    if (!this.localContractKitSigner) {
+      throw new Error('authentication needed: password or unlock')
+    }
+    return this.localContractKitSigner
   }
 
   /**
-   * Get the local signer. Throws if not unlocked.
+   * Get the unlocked ethers wallet. Throws if not unlocked.
    */
-  protected get localSigner(): LocalSigner {
+  get unlockedEthersWallets() {
     if (!this.isUnlocked()) {
-      this.unlockedLocalSigner = null
+      this.localContractKitSigner = null
```

Comment:
nit: this repeats the first part of `unlockedContractKitSigner` and could be made more DRY